src/main/java/org/cbioportal/security/config/OAuth2SecurityConfig.java

```diff
@@ -1,110 +1,156 @@
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper;
+import org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
 import org.springframework.security.oauth2.core.oidc.user.OidcUserAuthority;
 import org.springframework.security.oauth2.core.user.OAuth2UserAuthority;
-import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.authentication.HttpStatusEntryPoint;
-import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.authentication.logout.LogoutSuccessHandler;
 
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 
 @Configuration
-@EnableWebSecurity
 // add new chain after api-filter chain (at position -2), but before the default fallback chain 
 @Order(SecurityProperties.BASIC_AUTH_ORDER - 1)
+@ConditionalOnProperty(value = "authenticate", havingValue = "oauth2")
 public class OAuth2SecurityConfig {
-
-    
-    @Value("${spring.security.oauth2.roles-path.client-id:}")
-    private String clientId;
-    
-    @Value("${spring.security.oauth2.resourceserver.jwt.issuer-uri:}")
-    private String jwtResourceServerUri;
     
-    @Bean
-    @ConditionalOnProperty(value = "authenticate", havingValue = "oauth2")
-    public SecurityFilterChain oAuth2filterChain(HttpSecurity http) throws Exception {
-
-        http.authorizeHttpRequests(auth -> 
-                auth.requestMatchers("/api/health", "/login", "/images/**").permitAll()
-                    .anyRequest().authenticated())
-            .oauth2Login(oauth -> oauth.loginPage("/login"))
-            .logout((logout) -> logout.logoutSuccessUrl("/login?logout_success"))
-            .exceptionHandling(eh ->
-                eh.defaultAuthenticationEntryPointFor(new HttpStatusEntryPoint(HttpStatus.UNAUTHORIZED), AntPathRequestMatcher.antMatcher("/api/**")))
-            .csrf(AbstractHttpConfigurer::disable);
-        
-        if(!Objects.isNull(this.jwtResourceServerUri) && !this.jwtResourceServerUri.isEmpty()) {
-            http.oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
-        }
-        return http.build();
-    }
+    // TODO - add this to portal.properties.EXAMPLE
+    // TODO - discuss changing this to /logout (Spring Security default) with Aaron
+    @Value("${oauth2.logout.url:/j_spring_security_logout}")
+    private String logoutUrl;
```

Comment:
As per yesterday's discussion I think this is already `/logout` now right? CC @haynescd

Reply:
Indeed!